### ISSUE TITLE
MacOS build fail: esy bootstrap

### ISSUE THREAD

**Issue by wisonye** [onivim/oni2#3494]
MacOS info:

```bash
System Software Overview:

  System Version: macOS 10.14.6 (18G4032)
  Kernel Version: Darwin 18.7.0
```

Just follow the official doc to compile, but failed at `esy bootstrap`, here is the entire console log:

```bash
 I  wison | /Users/wison/temp/oni2   esy bootstrap

info building @esy-ocaml/substs@0.0.1@d41d8cd9
info building ocaml@4.10.2000@d41d8cd9
info building yarn-pkg-config@github:esy-ocaml/yarn-pkg-config#db3a0b63883606dd57c54a7158d560d6cba8cd79@d41d8cd9
info building esy-angle-prebuilt@1.0.0@d41d8cd9
info building esy-cmake@0.3.5@d41d8cd9
info building esy-oniguruma@github:onivim/esy-oniguruma#4698ce4@d41d8cd9
info building @esy-ocaml/substs@0.0.1@d41d8cd9: done
info building esy-sdl2@2.0.10008@d41d8cd9
info building esy-angle-prebuilt@1.0.0@d41d8cd9: done
info building @esy-cross/ninja-build@1.8.2001@d41d8cd9
info building esy-cmake@0.3.5@d41d8cd9: done
info building @revery/esy-cmake@0.3.5001@d41d8cd9
info building @esy-cross/ninja-build@1.8.2001@d41d8cd9: done
info building esy-nasm@github:revery-ui/esy-nasm#64a802b@d41d8cd9
info building esy-nasm@github:revery-ui/esy-nasm#64a802b@d41d8cd9: done
info building esy-tree-sitter@1.4.1@d41d8cd9
info building esy-tree-sitter@1.4.1@d41d8cd9: done
info building libvim@8.10869.87@d41d8cd9
info building esy-oniguruma@github:onivim/esy-oniguruma#4698ce4@d41d8cd9: done
info building esy-fzy@github:bryphe/esy-fzy#301dbf6@d41d8cd9
info building esy-fzy@github:bryphe/esy-fzy#301dbf6@d41d8cd9: done
info building @revery/esy-harfbuzz@2.6.8002@d41d8cd9
info building yarn-pkg-config@github:esy-ocaml/yarn-pkg-config#db3a0b63883606dd57c54a7158d560d6cba8cd79@d41d8cd9: done
info building @revery/esy-libvterm@1.0.4@d41d8cd9
info building @revery/esy-libvterm@1.0.4@d41d8cd9: done
info building @opam/base-bigarray@opam:base@b03491b0
info building @opam/base-bigarray@opam:base@b03491b0: done
info building @opam/base-threads@opam:base@36803084
info building @opam/base-threads@opam:base@36803084: done
info building @opam/base-unix@opam:base@87d0b2eb
info building @opam/base-unix@opam:base@87d0b2eb: done
info building esy-macdylibbundler@0.4.5001@d41d8cd9
info building esy-macdylibbundler@0.4.5001@d41d8cd9: done
info building esy-freetype2@2.9.1008@d41d8cd9
info building esy-freetype2@2.9.1008@d41d8cd9: done
info building @opam/conf-pkg-config@opam:1.2@d86c8f53
info building @opam/conf-pkg-config@opam:1.2@d86c8f53: done
info building libvim@8.10869.87@d41d8cd9: done
info building esy-sdl2@2.0.10008@d41d8cd9: done
info building ocaml@4.10.2000@d41d8cd9: done
info building @opam/dune@opam:2.8.5@83fb0224
info building @opam/ocamlfind@opam:1.9.1@b748edf6
info building @opam/cmdliner@opam:1.0.4@93208aac
info building @opam/ocamlbuild@opam:0.14.0@6ac75d03
info building @opam/cmdliner@opam:1.0.4@93208aac: done
info building @opam/seq@opam:base@d8d7de1d
info building @opam/seq@opam:base@d8d7de1d: done
info building @opam/ocamlfind@opam:1.9.1@b748edf6: done
info building @opam/base-bytes@opam:base@19d0c2ff
info building @opam/base-bytes@opam:base@19d0c2ff: done
info building @opam/ocamlbuild@opam:0.14.0@6ac75d03: done
info building @opam/topkg@opam:1.0.3@e4e10f1c
info building @opam/uchar@opam:0.0.2@c8218eea
info building @opam/uchar@opam:0.0.2@c8218eea: done
info building @opam/topkg@opam:1.0.3@e4e10f1c: done
info building @opam/astring@opam:0.8.5@1300cee8
info building @opam/uutf@opam:1.0.2@4440868f
info building @opam/uutf@opam:1.0.2@4440868f: done
info building @opam/uucp@opam:13.0.0@e9b515e0
info building @opam/astring@opam:0.8.5@1300cee8: done
info building @opam/fpath@opam:0.7.3@674d8125
info building @opam/fpath@opam:0.7.3@674d8125: done
info building @opam/dune@opam:2.8.5@83fb0224: done
info building @opam/omd@github:ocaml/omd:omd.opam#1535e3c@d41d8cd9
info building @opam/cppo@opam:1.6.7@c28ac3ae
info building @opam/bigarray-compat@opam:1.0.0@3a87ad65
info building @opam/bigarray-compat@opam:1.0.0@3a87ad65: done
info building @opam/ocaml-syntax-shims@opam:1.0.0@a9aa3bfa
info building @opam/cppo@opam:1.6.7@c28ac3ae: done
info building @opam/result@opam:1.5@6b753c82
info building @opam/ocaml-syntax-shims@opam:1.0.0@a9aa3bfa: done
info building @opam/camomile@opam:1.0.2@40411a6b
info building @opam/result@opam:1.5@6b753c82: done
info building @opam/decoders@opam:0.6.0@daf00a77
info building @opam/decoders@opam:0.6.0@daf00a77: done
info building @opam/easy-format@opam:1.3.2@0484b3c4
info building @opam/easy-format@opam:1.3.2@0484b3c4: done
info building @opam/fix@opam:20201120@5c318621
info building @opam/fix@opam:20201120@5c318621: done
info building @opam/menhirLib@opam:20210419@e9618e7a
info building @opam/omd@github:ocaml/omd:omd.opam#1535e3c@d41d8cd9: done
info building @opam/menhirSdk@opam:20210419@8a5d1ff3
info building @opam/menhirSdk@opam:20210419@8a5d1ff3: done
info building @opam/ppx_derivers@opam:1.2.1@ecf0aa45
info building @opam/menhirLib@opam:20210419@e9618e7a: done
info building @opam/dune-private-libs@opam:2.5.1@60c1661f
info building @opam/ppx_derivers@opam:1.2.1@ecf0aa45: done
info building @opam/stdlib-shims@opam:0.3.0@0d088929
info building @opam/stdlib-shims@opam:0.3.0@0d088929: done
info building @opam/mmap@opam:1.1.0@b85334ff
info building @opam/mmap@opam:1.1.0@b85334ff: done
info building @opam/psq@opam:0.2.0@247756d4
info building @opam/psq@opam:0.2.0@247756d4: done
info building @opam/integers@opam:0.4.0@f7acfaeb
info building @opam/integers@opam:0.4.0@f7acfaeb: done
info building @opam/markup@opam:0.8.2@53ccafce
info building @opam/markup@opam:0.8.2@53ccafce: done
info building @opam/csexp@opam:1.5.1@f2f16ef6
info building @opam/csexp@opam:1.5.1@f2f16ef6: done
info building @opam/dune-build-info@opam:2.8.5@1dae72be
info building @opam/dune-build-info@opam:2.8.5@1dae72be: done
info building @opam/sexplib0@opam:v0.14.0@ddeb6438
info building @opam/sexplib0@opam:v0.14.0@ddeb6438: done
info building @opam/re@opam:1.9.0@d4d5e13d
info building @opam/dune-private-libs@opam:2.5.1@60c1661f: done
info building @opam/ocaml-compiler-libs@opam:v0.12.3@f0f069bd
info building @opam/camomile@opam:1.0.2@40411a6b: done
info building @opam/jane-street-headers@opam:v0.14.0@59432b6a
info building @opam/jane-street-headers@opam:v0.14.0@59432b6a: done
info building @opam/octavius@opam:1.2.2@2c6624f5
info building @opam/ocaml-compiler-libs@opam:v0.12.3@f0f069bd: done
info building @opam/bigstringaf@opam:0.7.0@4784da9b
info building @opam/re@opam:1.9.0@d4d5e13d: done
info building @opam/merlin-extend@opam:0.6@404f814c
info building @opam/bigstringaf@opam:0.7.0@4784da9b: done
info building @opam/ocplib-endian@opam:1.1@84c1ca88
info building @opam/merlin-extend@opam:0.6@404f814c: done
info building @opam/ptime@opam:0.8.5@0051d642
info building @opam/octavius@opam:1.2.2@2c6624f5: done
info building @opam/rresult@opam:0.6.0@4b185e72
info building @opam/ocplib-endian@opam:1.1@84c1ca88: done
info building @opam/biniou@opam:1.2.1@d7570399
info building @opam/rresult@opam:0.6.0@4b185e72: done
info building @opam/menhir@opam:20210419@f3bb7fec
info building @opam/biniou@opam:1.2.1@d7570399: done
info building @opam/ocaml-migrate-parsetree@opam:1.8.0@caf9847c
info building @opam/ptime@opam:0.8.5@0051d642: done
info building @opam/fmt@opam:0.8.9@e0843a5b
info building @opam/uucp@opam:13.0.0@e9b515e0: done
info building @opam/lru@github:bryphe/lru:lru.opam#2708c70@d41d8cd9
info building @opam/lru@github:bryphe/lru:lru.opam#2708c70@d41d8cd9: done
info building @opam/ctypes@opam:0.15.1@9c7ab4ec
info building @opam/fmt@opam:0.8.9@e0843a5b: done
info building @opam/dune-configurator@opam:2.5.1@aeb9d8d5
info building @opam/dune-configurator@opam:2.5.1@aeb9d8d5: done
info building @opam/charInfo_width@opam:1.1.0@4296bdfe
info building @opam/charInfo_width@opam:1.1.0@4296bdfe: done
info building @opam/tyxml@opam:4.5.0@4676c9e9
info building @opam/menhir@opam:20210419@f3bb7fec: done
info building @opam/angstrom@opam:0.15.0@48ede9cb
info building @opam/tyxml@opam:4.5.0@4676c9e9: done
info building @opam/yojson@github:onivim/yojson:yojson.opam#f480aef@d41d8cd9
info building @opam/angstrom@opam:0.15.0@48ede9cb: done
info building @opam/uuseg@opam:13.0.0@f60712a7
info building @opam/ocaml-migrate-parsetree@opam:1.8.0@caf9847c: done
info building @opam/lwt@opam:4.5.0@457db00f
info building @opam/yojson@github:onivim/yojson:yojson.opam#f480aef@d41d8cd9: done
info building reason-native-crash-utils@github:onivim/reason-native-crash-utils#a80a9d9@d41d8cd9
info building @opam/uuseg@opam:13.0.0@f60712a7: done
info building @onivim/reason-native-crash-utils@1.0.2@d41d8cd9
info building reason-native-crash-utils@github:onivim/reason-native-crash-utils#a80a9d9@d41d8cd9: done
info building @opam/base@opam:v0.14.1@d14008e2
info building @onivim/reason-native-crash-utils@1.0.2@d41d8cd9: done
info building @opam/reason@opam:3.7.0@191be014
info building @opam/ctypes@opam:0.15.1@9c7ab4ec: done
info building @esy-ocaml/reason@3.7.0@d41d8cd9
info building @opam/lwt@opam:4.5.0@457db00f: done
info building @opam/atd@opam:2.2.1@071ab6bd
info building @opam/atd@opam:2.2.1@071ab6bd: done
info building @opam/junit@opam:2.0.2@f4402cc0
info building @opam/junit@opam:2.0.2@f4402cc0: done
info building @opam/odoc@opam:1.5.2@94f47c8b
info building @opam/base@opam:v0.14.1@d14008e2: done
info building @opam/printbox@opam:0.5@82f5d436
info building @opam/printbox@opam:0.5@82f5d436: done
info building @opam/semver2@opam:1.2.0@ced7107e
info building @opam/semver2@opam:1.2.0@ced7107e: done
info building @opam/ppxlib@opam:0.15.0@04a04df2
info building @opam/odoc@opam:1.5.2@94f47c8b: done
info building @opam/ppx_tools_versioned@opam:5.4.0@4ab57069
info building @opam/ppx_tools_versioned@opam:5.4.0@4ab57069: done
info building @opam/atdgen-runtime@opam:2.2.1@6a3a6395
info building @opam/atdgen-runtime@opam:2.2.1@6a3a6395: done
info building @opam/dot-merlin-reader@opam:4.1@120afa42
info building @esy-ocaml/reason@3.7.0@d41d8cd9: done
info building @opam/decoders-yojson@opam:0.4.0@cb39dea6
info building @opam/decoders-yojson@opam:0.4.0@cb39dea6: done
info building @opam/ppx_yojson_conv_lib@opam:v0.14.0@116b53d6
info building @opam/dot-merlin-reader@opam:4.1@120afa42: done
info building @opam/luv@github:bryphe/luv:luv.opam#8e9f2b0@d41d8cd9
info building @opam/ppx_yojson_conv_lib@opam:v0.14.0@116b53d6: done
info building @opam/logs@opam:0.7.0@1d03143e
info building @opam/reason@opam:3.7.0@191be014: done
info building @opam/stdio@opam:v0.14.0@a624e254
info building @opam/stdio@opam:v0.14.0@a624e254: done
info building @opam/lwt_ppx@opam:2.0.1@edf5ad68
info building @opam/logs@opam:0.7.0@1d03143e: done
info building @opam/atdgen@opam:2.2.1@d73fda11
info building @opam/ppxlib@opam:0.15.0@04a04df2: done
info building rebez@github:jchavarri/rebez#03fa3b7@d41d8cd9
info building @opam/atdgen@opam:2.2.1@d73fda11: done
info building flex@github:bryphe/flex#5e19b05@d41d8cd9
info building rebez@github:jchavarri/rebez#03fa3b7@d41d8cd9: done
info building @reason-native/pastel@0.3.0@d41d8cd9
info building @opam/lwt_ppx@opam:2.0.1@edf5ad68: done
info building @reason-native/console@0.1.0@d41d8cd9
info building @reason-native/console@0.1.0@d41d8cd9: done
info building @opam/ocaml-lsp-server@github:ocaml/ocaml-lsp:ocaml-lsp-server.opam#966a28f@d41d8cd9
info building @reason-native/pastel@0.3.0@d41d8cd9: done
info building reperf@1.5.1@d41d8cd9
info building reperf@1.5.1@d41d8cd9: done
info building isolinear@github:revery-ui/isolinear#049ad1e@d41d8cd9
info building isolinear@github:revery-ui/isolinear#049ad1e@d41d8cd9: done
info building rench@1.10.0@d41d8cd9
info building flex@github:bryphe/flex#5e19b05@d41d8cd9: done
info building @opam/fp@github:bryphe/reason-native:fp.opam#e16590c@d41d8cd9
info building rench@1.10.0@d41d8cd9: done
info building @opam/ocamlformat@opam:0.15.0@8e036963
info building @revery/esy-harfbuzz@2.6.8002@d41d8cd9: done
info building @revery/timber@2.0.0@d41d8cd9
info building @opam/fp@github:bryphe/reason-native:fp.opam#e16590c@d41d8cd9: done
info building @opam/bos@opam:0.2.0@df49e63f
info building @revery/timber@2.0.0@d41d8cd9: done
info building @opam/ppx_let@opam:v0.14.0@eb9b93e0
info building @opam/ppx_let@opam:v0.14.0@eb9b93e0: done
info building @opam/ppx_optcomp@opam:v0.14.0@47cec200
info building @opam/bos@opam:0.2.0@df49e63f: done
info building @opam/ppx_js_style@opam:v0.14.0@10b020a8
info building @opam/ppx_optcomp@opam:v0.14.0@47cec200: done
info building @opam/ppx_enumerate@opam:v0.14.0@63db8245
info building @opam/ppx_js_style@opam:v0.14.0@10b020a8: done
info building @opam/ppx_sexp_conv@opam:v0.14.1@5fc4da3c
info building @opam/ppx_enumerate@opam:v0.14.0@63db8245: done
info building @opam/ppx_here@opam:v0.14.0@5ccc1c01
info building @opam/ppx_here@opam:v0.14.0@5ccc1c01: done
info building @opam/ppx_compare@opam:v0.14.0@615de7a6
info building @opam/luv@github:bryphe/luv:luv.opam#8e9f2b0@d41d8cd9: done
info building @opam/ppx_cold@opam:v0.14.0@345dec7c
info building @opam/ppx_sexp_conv@opam:v0.14.1@5fc4da3c: done
info building @opam/ppx_deriving@opam:5.1@089b343d
info building @opam/ppx_compare@opam:v0.14.0@615de7a6: done
info building @brisk/brisk-reconciler@github:briskml/brisk-reconciler#c9d5c4c@d41d8cd9
info building @opam/ppx_cold@opam:v0.14.0@345dec7c: done
info building refmterr@3.3.2@d41d8cd9
info building refmterr@3.3.2@d41d8cd9: done
info building @reason-native/cli@0.0.1-alpha@d41d8cd9
info building @brisk/brisk-reconciler@github:briskml/brisk-reconciler#c9d5c4c@d41d8cd9: done
info building @reason-native/file-context-printer@0.0.3@d41d8cd9
info building @opam/ppx_deriving@opam:5.1@089b343d: done
info building @opam/fs@github:bryphe/reason-native:fs.opam#e16590c@d41d8cd9
info building @reason-native/file-context-printer@0.0.3@d41d8cd9: done
info building @opam/dir@github:bryphe/reason-native:dir.opam#e16590c@d41d8cd9
info building @reason-native/cli@0.0.1-alpha@d41d8cd9: done
info building @opam/ppx_hash@opam:v0.14.0@8e9618e4
info building @opam/dir@github:bryphe/reason-native:dir.opam#e16590c@d41d8cd9: done
info building @opam/ppx_assert@opam:v0.14.0@877b5900
info building @opam/fs@github:bryphe/reason-native:fs.opam#e16590c@d41d8cd9: done
info building @opam/ppx_deriving_yojson@opam:3.6.1@faf11a7c
info building @opam/ppx_hash@opam:v0.14.0@8e9618e4: done
info building @reason-native/rely@3.2.1@d41d8cd9
info building @opam/ocamlformat@opam:0.15.0@8e036963: done
info building @opam/ppx_base@opam:v0.14.0@b4702ed9
info building @opam/ppx_deriving_yojson@opam:3.6.1@faf11a7c: done
info building @opam/ppx_assert@opam:v0.14.0@877b5900: done
info building @opam/jst-config@opam:v0.14.0@d0d7469e
info building @reason-native/rely@3.2.1@d41d8cd9: done
info building @opam/ocaml-lsp-server@github:ocaml/ocaml-lsp:ocaml-lsp-server.opam#966a28f@d41d8cd9: done
info building @opam/ppx_base@opam:v0.14.0@b4702ed9: done
info building @opam/jst-config@opam:v0.14.0@d0d7469e: done
info building @opam/time_now@opam:v0.14.0@5e4046b3
info building @opam/time_now@opam:v0.14.0@5e4046b3: done
info building @opam/ppx_inline_test@opam:v0.14.1@2e4fdd8d
info building @opam/ppx_inline_test@opam:v0.14.1@2e4fdd8d: done
info building reason-fzy@github:CrossR/reason-fzy#485cae1@d41d8cd9
info building reason-fzy@github:CrossR/reason-fzy#485cae1@d41d8cd9: done
info building @revery/esy-cmake@0.3.5001@d41d8cd9: done
info building esy-libjpeg-turbo@github:revery-ui/libjpeg-turbo#dbb3dd5@d41d8cd9
info building esy-libjpeg-turbo@github:revery-ui/libjpeg-turbo#dbb3dd5@d41d8cd9: done
info building esy-skia@github:revery-ui/esy-skia#91c98f6@d41d8cd9
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: esy-skia@github:revery-ui/esy-skia#91c98f6
    # esy-build-package: pwd: /Users/wison/.esy/3/b/esy_skia-c3eadd32
    # esy-build-package: running: 'bash' './esy/build.sh' 'darwin' '/Users/wison/.esy/3__________________________________________________________________/i/esy_libjpeg_turbo-dc3b4679'
    fatal: remote error: Git repository not found
    buildtools                           @ 505de88083136eefd056e5ee4ca0f01fe9b33de8
    Traceback (most recent call last):
      File "tools/git-sync-deps", line 254, in <module>
        exit(main(sys.argv[1:]))
      File "tools/git-sync-deps", line 246, in main
        git_sync_deps(deps_file_path, argv, verbose)
      File "tools/git-sync-deps", line 218, in git_sync_deps
        git_checkout_to_directory(*args)
      File "tools/git-sync-deps", line 132, in git_checkout_to_directory
        [git, 'clone', '--quiet', repo, directory])
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 190, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['git', 'clone', '--quiet', 'https://skia.googlesource.com/common.git', 'common']' returned non-zero exit status 128
    llvm toolset-7.0 does not need to be manually activated
    ./esy/build.sh: line 44: bin/gn: No such file or directory
    error: command failed: 'bash' './esy/build.sh' 'darwin' '/Users/wison/.esy/3__________________________________________________________________/i/esy_libjpeg_turbo-dc3b4679' (exited with 255)
    esy-build-package: exiting with errors above...

  building esy-skia@github:revery-ui/esy-skia#91c98f6
esy: exiting due to errors above
```

</br>

As it mentions `fatal: remote error: Git repository not found`, then I did give it a try on the missing git repo, but no luck:

```bash
 git clone https://skia.googlesource.com/common.git
Cloning into 'common'...
fatal: remote error: Git repository not found
```

</br>

And then I open the link, it seems the `common` repo really doesn't exist?:)

<img width="607" alt="Screen Shot 2021-05-04 at 4 28 39 PM" src="https://user-images.githubusercontent.com/3477321/116961127-92d07780-acf6-11eb-933f-89ca53b882bb.png">

**Comment by jlorince** [comment]
Encountering the same issue, fwiw

**Comment by wisonye** [comment]
@bryphe  Yup, thanks for your help, feel free to close this issue, then we know that your PR has been merged and we can try again:) Thank a lot:)

**Comment by bryphe** [comment]
Thanks for the logging the issue @wisonye ! Was hitting this on my Manjaro development box too - should be fixed once #3500
